tests/components/mqtt/test_sensor.py

```diff
@@ -80,6 +80,58 @@ async def test_setting_sensor_value_via_mqtt_message(hass, mqtt_mock):
     assert state.attributes.get("unit_of_measurement") == "fav unit"
 
 
+@pytest.mark.parametrize(
+    "device_class,native_value,state_value,log",
+    [
+        (sensor.DEVICE_CLASS_DATE, "2021-11-18", "2021-11-18", False),
+        (sensor.DEVICE_CLASS_DATE, "invalid", STATE_UNKNOWN, True),
+        (
+            sensor.DEVICE_CLASS_TIMESTAMP,
+            "2021-11-18T20:25:00",
+            "2021-11-18T20:25:00",
+            False,
+        ),
+        (
+            sensor.DEVICE_CLASS_TIMESTAMP,
+            "2021-11-18 20:25:00",
```

Comment:
This is a naive timestamp, should we reject it in `SensorEntity`?

Reply:
Yup: #59971